File: assignment/Assessment/assessment/Nokia.java
import java.util.Scanner;
public class Nokia{
	public static void main(String[] args){

System.out.println("Welcome to your Nokia phone\n List of menu functions\n1 ->Phone book\n 2->Messages\n 3->Chat\n 4->Call register\n 5->Tones\n 6->Settings\n 7->Call divert\n 8->Games\n 9->Calculator\n 10->Reminders\n 11->Clock\n 12->Profiles\n 13->SIM services\n "); 

System.out.println("menu");

Scanner input=new Scanner(System.in);
System.out.println("Input a number");
int choice =input.nextInt();

switch (choice) {
case 1:
	System.out.println("Phone book\n 1: Search\n 2: Service Nos.\n 3: Add name\n 4: Erase\n 5: Edit\n 6: Assign tone\n 7: Send b'card\n 8: Options\n 9: Speed dials\n 10: Voice tags ");

System.out.println("phone book");
Scanner menu = new Scanner(System.in);
System.out.println("Input a number:");
int Phonebook = input.nextInt();


switch (Phonebook) {
	case 1:
		System.out.println("1:Search"); 
			break;
	case 2:
		System.out.println("2:Service Nos");
			 break;
	case 3:
		System.out.println("3:Add Name");
			 break;
	case 4:
		System.out.println("4:Erase"); 
			break;
	case 5:
		System.out.println("5:Edit");
			 break;
	case 6:
		System.out.println("6:Assign Tone");
			 break;
	case 7:
		System.out.println("7:Send b'card"); 
			break;
	case 8:
		System.out.println("Options\ninput an option\n 1. Types of view\n 2. Memory status");
			Scanner Input = new Scanner(System.in);
			int Options =Input.nextInt();
			switch(Options) {
		case 1:
		System.out.println("Types Of View");
			 break;
		case 2:
		System.out.println("Memory Status");
			 break;
			}
			break;
	case 9:
		System.out.println("Speed Dials"); 
			break;
	case 10:
		System.out.println("Voice Tags"); 
			break;

	default: System.out.print(" Invalid ");
			break;

		}
		
			break;
	case 2:
		System.out.println("Messages\n1:Write messages\n2:Inbox\n3:Outbox\n4:Picture messages\n5:Templates\n6:Smileys\n7:Message settings\n8:Info service\n9:Voice mailbox number\n10:Service command editor");

System.out.println("messsages");
Scanner map = new Scanner(System.in);
System.out.println("Input a number:");
int Messages = map.nextInt();

switch (Messages) {

	case 1:
		System.out.println("Write messages"); 
			break;
	case 2:
		System.out.println("Inbox");
			 break;
	case 3:
		System.out.println("Outbox");
			 break;
	case 4:                                    
		System.out.println("Picture messages"); 
			break;
	case 5:
		System.out.println("Templates");
			 break;
	case 6:
		System.out.println("Smileys");
			 break;
	case 7:
System.out.println("Message settings"); 
System.out.println("input a number");
System.out.println("1.Set 1");
System.out.println("2.Common");

Scanner submenu = new Scanner(System.in);
int MessagesSettings = submenu.nextInt();
switch(MessagesSettings){
	case 1:
		System.out.println("Input an Options:");
		System.out.println("1.Message Centre Number");
		System.out.println("2.Message Sent As");
		System.out.println("3.Message Validity");

		Scanner menumap = new Scanner(System.in);
		int menumapnumber = menumap.nextInt();
		switch(menumapnumber){
	case 1:	
		System.out.println("Message Centre Number"); 
			break;
	case 2:
		System.out.println("Message Sent As");
			 break;
	case 3:
		System.out.println("Message Validity"); 
			break;
}
			break;
	case 2:
		System.out.println("Common");
		System.out.println("Input a number:");
		System.out.println("1.Delivery Reports");
		System.out.println("2.Reply Via Same Centre");
		System.out.println("3.Character Support");

		Scanner menumap2 = new Scanner(System.in);
		int menumapnumber2 = menumap2.nextInt();
		switch (menumapnumber2) {
	case 1:	
		System.out.println("Delivery Reports");
			 break;
	case 2:
		System.out.println("Reply Via Same Centre");
			 break;
	case 3:
		System.out.println("Character Support");
			 break;
}

			break;
} 
			break;

	case 8:
		System.out.println("Info sevice");
			break;
	case 9:
		System.out.println("Voice mailbox number");
			break;
	case 10:
		System.out.println("Service command editor");
			break;
		default: System.out.print(" Invalid ");
}
			break;
	

	case 3:
		System.out.println("Chat");
			break;
		
	case 4:
		System.out.println(" call register/n 1: missed calls/n 2: received calls/n 3: Dialled number/n 4: Erase recent call lists/n 5: Show call duration/n 6: Show call costs/n 7: call cost settings/n 8: prepaid credit ");

	System.out.println("callregister");
	Scanner calls = new Scanner(System.in);
	System.out.println("Input a number:");
	int callregister = calls.nextInt();

	switch (callregister) {

		case 1:	
			System.out.println("missed calls");
				break;
		case 2:
			System.out.println("received calls");
				break;
		case 3:
			System.out.println("dialled numbers");
				break;
		case 4:
			System.out.println("erase recent call lists");

				break;
		case 5:
			System.out.println("1. last call duration/n 2. All calls'duration/n 3. Received calls'duration/n 4. dialled calls duratio/n  5. clear timers ");

	System.out.println("Call duration");
	Scanner call = new Scanner(System.in);
	System.out.println("Input a number");
	int showcallduration = call.nextInt();

	switch (showcallduration) {

		case 1:	
			System.out.println("last call duration");
				break;
		case 2:
			System.out.println("all calls'duration");
				break;
		case 3:
			System.out.println("received calls'duration");
				break;
		case 4:
			System.out.println("dialled calls'duration");
				break;
		case 5:
			System.out.println("clear timers");
				break;			
}
				break;
		case 6:
			System.out.println(" last call cost/n 2. All calls'cost /n3. Clear counters ");

System.out.println("call cost");
	Scanner cost = new Scanner(System.in);
	System.out.println("Input a number");
	int ShowCallCost = cost.nextInt();

	switch (ShowCallCost) {
	
		case 1: 
			System.out.println("Last call cost");
				break;
		case 2:
			System.out.println("All calls'cost");
				break;
		case 3:
			System.out.println("Clear counters");
}
				break;
		case 7: 
			System.out.println("call costs settings\n 1.call cost limit\n 2.show costs in");
System.out.println("cost settings");
	Scanner costs = new Scanner(System.in);
	System.out.println("Input a number");
	int CallCostSettings = costs.nextInt();

	switch (CallCostSettings) {

		case 1:
			System.out.println("1:Call cost limit");
				break;
		case 2:
			System.out.println("2:Show costs in");
}
				break;
		case 8:
			System.out.println("3:Prepaid credit");
				break;

		default: System.out.print(" Invalid ");
				break;

}
			break;
}switch (choice) {
    case 5:
        System.out.println("Tones");
        System.out.println("1. Ringing tone\n 2. Ringing volume\n 3. Incoming call alert\n 4. Composer\n 5. Message alert tone\n 6. Keypad tones\n 7. Warning and game tones\n 8. Vibrating alert\n 9. Screen saver");

        Scanner tone = new Scanner(System.in);
        System.out.println("Input a number");
        int tones = tone.nextInt();
        switch (tones) {
            case 1:
                System.out.println("Ringing tone");
                break;
            case 2:
                System.out.println("Ringing volume");
                break;
            case 3:
                System.out.println("Incoming Call Alert");
                break;
            case 4:
                System.out.println("Composer");
                break;
            case 5:
                System.out.println("Message Alert Tone");
                break;
            case 6:
                System.out.println("Keypad Tones");
                break;
            case 7:
                System.out.println("Warning And Game Tones");
                break;
            case 8:
                System.out.println("Vibrating Alert");
                break;
            case 9:
                System.out.println("Screen Saver");
                break;
        }
        break;
    case 6:
        System.out.println("Call Settings");
        System.out.println("1. Automatic redial\n 2. Speed dialling\n 3. Call waiting options\n 4.Own number sending\n 5. Phone line in use\n 6. Automatic answer");

        Scanner caller = new Scanner(System.in);
        System.out.println("Input a number");
        int call = caller.nextInt();
        switch (call) {
            case 1:
                System.out.println("Automatic redial");
                break;
            case 2:
                System.out.println("Speed dialling");
                break;
            case 3:
                System.out.println("Call waiting options");
                break;
            case 4:
                System.out.println("Own number sending");
                break;
            case 5:
                System.out.println("Phone line in use");
                break;
            case 6:
                System.out.println("Automatic answer");
                break;
        }
        break;
    case 7:
        System.out.println("Call divert");
        break;
    case 8:
        System.out.println("Games");
        break;
    case 9:
        System.out.println("Calculator");
        break;
    case 10:
        System.out.println("Reminder");
        break;
	default: System.out.print(" Invalid ");
		break;

    case 11:
        System.out.println("Clock");
        System.out.println("1. Alarm clock\n 2. Clock settings\n 3. Date setting\n 4. Stopwatch\n 5. Countdown timer\n 6. Auto update of date and time");

        Scanner alarm = new Scanner(System.in);
        System.out.println("Input a number");
        int clock = alarm.nextInt();

        switch (clock) {
            case 1:
                System.out.println("Alarm Clock");
                break;
            case 2:
                System.out.println("Clock Setting");
                break;
            case 3:
                System.out.println("Date Setting");
                break;
            case 4:
                System.out.println("Stopwatch");
                break;
            case 5:
                System.out.println("Countdown Timer");
                break;
            case 6:
                System.out.println("Auto update of date and time");
                break;
	   default: System.out.print(" Invalid ");

        } 
        break;
    case 12:
        System.out.println("Profiles");
        break;
    case 13:
        System.out.println("Sim Services");
        break;
		}
	}
}

File: assignment/KATA/Nokia.py

def Nokia():
    menu_option = """
        1 -> PhoneBook
	2 -> Messages
	3 -> Chat
	4 -> Call Register
	5 ->Tones
	6 ->Settings
	7 -> Call Divert
	8 -> Games
	9 -> Calculator
	10 -> Reminders
	11 -> Clock
	12 -> Profiles
	13 -> SIM Services

    """
    print(menu_option)
    option = int(input("Enter A Number:"))

    match option:
        case 1:
            print("phonebook")
            def phonebook():
                menu = """
                1. Search
                2.Service Nos
                3. Add Name
                4. Erase
                5.Edit
                6.Assign Tone
                7. Send b'card
                8.Options
                9. Speed DIals
                10. Voice Tags
                11. Go Back
                """
                print(menu)
                input = int(input("Choose: "))
                match input:
                    case 1: print("Search")
                    case 2: print("Service Nos")
                    case 3: print("Add Name")
                    case 4: print("Erase")
                    case 5: print("Edit")
                    case 6: print("Assign Tone")
                    case 7: print("Send b'card")
                    case 8:
                        print("Options")
                        def option():
                            option = """
                            1. Types of Views
                            2. Memory Status
                            0. Back
                            """
                            print(option)
                            input = int(input("Choose: "))
                            match input:
                                case 1: print("Types Of Views")
                                case 2: print("Memory Status")
                                case 0:
                                    print("back")
                                    phonebook()
                                
                                
                        option()
                    case 9: print("Speed Dials")
                    case 10: print("Voice Tags")
                    case 11:
                        print("Go Back")
                        runMatch();
                        
                    
            phonebook()
        case 2:
            print("Messages")
            def messages():
                Messages = """
                1. Write Messages
                2. Inbox
                3. Outbox
                4.Picture
                5.Template
                6. Smileys
                7. Message Settings
                8. Info Service
                9. Voice Mailbox Number
                10. Service Command editor
                0. Go Back
                """
                print(Messages)
                inbox = int(input("Choose: "))
                match inbox:
                    case 1:print("Write Messages")
                    case 2:print("Inbox")
                    case 3:print("Outbox")
                    case 4:print("Pictures")
                    case 5:print("Template")
                    case 6:print("Smileys")
                    case 7:
                        print("Message Settings")
                        def message_settings():
                            Message_settings = """
                            1. Set
                            2.Common
                            0.Go BAck
                            """
                            print(option)
                            option = int(input("Choose: "))
                            match option:
                                case 1:
                                    print("Set")
                                    def set():
                                        sets = """
                                        1. Message centre number
                                        2. Messages sent as
                                        3. Message validity
                                        0. Back
                                        """
                                        print(sets)
                                        option = int(input("Choose: "))
                                        if option == 1: print("Message centre number")
                                        elif option == 2: print("Messages sent as")
                                        elif option == 3: print("Message validity")
                                        elif option == 0: print("Back")
                                        Message_settings()
                                        
                                    set()
                                case 2:
                                    print("Common")
                                    def common():
                                        commons = """
                                        1. Delivery reports
                                        2. Reply via same centre
                                        3. Character support
                                        0. Back
                                        """
                                        print(commons)
                                        common = int(input("Choose: "))
                                        if common == 1: print("Delivery reports")
                                        elif common == 2: print("Reply via same centre")
                                        elif common == 3: print("Character support")
                                        elif common == 0: print("Back")
                                        Message_settings()
                                    common()
                                case 0:
                                    print("Go back")
                                    messages()
        
                        Message_settings()
                    case 8: print("Info Service")
                    case 9: print("Voice mailbox")
                    case 10: print("service command ....")
                    case 0:
                        print("Go Back")
                        runMatch()
            messages()
        case 3: print("Chat")
        case 4:
            print("Call Register")
            call_register = """
            1. Missed calls
            2. Received calls
            3. Dialled numbers
            4. Erase recent call lists
            5. Show call duration
            6. Show call costs
            7. Call cost settings
            8. Prepaid credit
            """
            print(call_register)
            register = int(input("Choose: "))
            match register:
                case 1: print("Missed calls")
                case 2: print("Received calls")
                case 3: print("Dialled numbers")
                case 4: print("Erase recent call lists")
                case 5:
                    print("Show call duration")
                    def call_duration():
                        duration = """
                        1. Last Call Duration
                        2. All calls’ duration
                        3. Received calls’ duration
                        4. Dialled calls’ duration
                        5. Clear timers
                        """
                        print(duration)
                        duration = int(input("Choose: "))
                        if duration == 1: print("Last Call Duration")
                        elif duration == 2: print("All calls’ duration")
                        elif duration == 3: print("Received calls’ duration")
                        elif duration == 4: print("Dialled calls’ duration")
                        else : print("Clear timers")
                    call_duration()
                case 6:
                    print("Show call costs")
                    def call_costs():
                        call_cost = """
                        1.Last call cost
                        2. All calls’ cost
                        3.Clear counters
                        """
                        print(call_cost)
                        cost = int(input("Choose: "))
                        if cost == 1: print("Last call cost")
                        elif cost == 2: print("All calls’ cost")
                        else: print("Clear Counter")
                    call_costs()
                case 7:
                    print("Call cost")
                    def call_costs():
                        cost_settings = """
                        1. Call cost limit
                        2. Show costs in
                        """
                        print(cost_settings)
                        call_cost = int(input("Choose: "))
                        if call_cost == 1: print("Call cost limit")
                        else: print("Show costs in")
                    call_costs()
                case 8: print("Prepaid credit")
        case 5:
            print("Tone")
            tone = """
            1. Ringing tone
            2. Ringing volume
            3. Incoming call alert
            4. Composer
            5. Message alert tone
            6. Keypad tones
            7. Warning and game tones
            8. Vibrating alert
            9. Screen saver
            """
            print(tones)
            tones = int(input("Choose: "))
            match tones:
                case 1:print("Ringing tone")
                case 2:print("Ringing volume")
                case 3:print("Incoming call alert")
                case 4:print("Composer")
                case 5:print("Message alert tone")
                case 6:print("Keypad tones")
                case 7:print("Warning and game tones")
                case 8:print("Vibrating alert")
                case 9:print("Screen saver")

        case 6:
            print("Settings")
            def settings():
                setting = """
                1. Call settings
                2.Phone settings
                3. Security settings
                4. Restore factory settings
                """
                print(setting)
                setting =  int(input("Choose: "))
                match setting:
                    case 1:
                        print("Call Settings")
                        def calls():
                            calls = """
                            1. Automatic redial
                            2. Speed dialling
                            3. Call waiting options
                            4. Own number sending
                            5. Phone line in use
                            6. Automatic answer
                            """
                            print(call)
                            call = int(input("Enter: "))
                            match call:
                                case 1: print("Auto Redial")
                                case 2: print("Speed dialling")
                                case 3: print("Call waiting options")
                                case 4: print("Own number sending")
                                case 5: print("Phone line in use")
                                case 6: print("Automatic answer")
                        call_settings()
                    case 2:
                        print("Phone Settings")
                        def phone_settings():
                            phone = """
                            1. Language
                            2. Cell info display
                            3. Welcome note
                            4. Network selection
                            5. Lights 
                            6. Confirm SIM service actions
                            """
                            print(phone)
                            phone = int(input("Enter: "))
                            match phone:
                                case 1: print("Language")
                                case 2: print("Cell info display")
                                case 3: print("Welcome note")
                                case 4: print("Network selection")
                                case 5: print("Lights ")
                                case 6: print("Confirm SIM service actions")
                        phone_settings()
                    case 3:
                        print("Security Settings")
                        def securitys():
                            securitys = """
                            1. PIN code request
                            2. Call barring service
                            3. Fixed dialling
                            4. Closed user group
                            5. Phone security
                            6. Change access codes
                            """
                            print(security_seetings_menu)
                            securities = int(input("Enter: "))
                            match securities:
                                case 1: print("PIN code request")
                                case 2: print("Call barring service")
                                case 3: print("Fixed dialling")
                                case 4: print("Closed user group")
                                case 5: print("Phone security")
                                case 6: print("Change access codes")
                        security_settings()
                    case 4: print("Restore Factory code")
            settings()
        case 7: print("Call Divert")
        case 8: print("Games")
        case 9: print("Calculator")
        case 10: print("Reminders")
        case 11:
            print("Clock")
            def clock():
                clocks = """
                1. Alarm clock
                2. Clock settings
                3. Date setting
                4. Stopwatch
                5. Countdown timer
                6. Auto update of date and time
                """
                print(clocks)
                input = int(input("Choose: "))
                match input:
                    case 1: print("Alarm clock")
                    case 2: print("Clock settings")
                    case 3: print("Date setting")
                    case 4: print("Stopwatch")
                    case 5: print("Countdown timer")
                    case 6: print("Auto update of date and time")
            clock()
        case 12: print("Profiles")
        case 13: print("SIM Services")
Nokia():
                    
runMatch()
